pythran/dist.py

```diff
@@ -4,13 +4,48 @@
 '''
 
 import pythran.config as cfg
-import pythran.toolchain as tc
-
-from numpy.distutils.extension import Extension
 
 import os.path
 import os
 
+from numpy.distutils.extension import Extension
+
+from numpy.distutils.command import build_ext as build_ext0
+from distutils.command import build_ext as build_ext1
+
+for build_ext in (build_ext0, build_ext1):
+
+    LegacyBuildExt = build_ext.build_ext
+
+    class PythranBuildExt(LegacyBuildExt):
+
+        # workaround mismatch between distutils and numpy
+        # implementation
+        def swig_sources(self, sources, *args):
+            return sources
+
+        def build_extension(self, ext):
+            prev = {'preprocessor': None,
+                    'compiler_cxx': None,
+                    'compiler_so': None,
+                    'compiler': None,
+                    'linker_exe': None,
+                    'linker_so': None}
+            if getattr(ext, 'cxx', None) and 'CXX' not in os.environ:
+                for attr in prev.keys():
+                    prev[attr] = getattr(self.compiler, attr)[0]
+                    getattr(self.compiler, attr)[0] = ext.cxx
+
+                try:
+                    return LegacyBuildExt.build_extension(self, ext)
+                finally:
+                    for attr in prev.keys():
+                        getattr(self.compiler, attr)[0] = prev[attr]
+            else:
+                return LegacyBuildExt.build_extension(self, ext)
```

Comment:
Maybe use `super(PythranBuildExt, self).build_extension(self, ext)`  here and above?